ddtrace/contrib/pytest/_atr_utils.py

```diff
@@ -29,41 +29,69 @@ class _ATR_RETRY_OUTCOMES:
     ATR_FINAL_FAILED = "dd_atr_final_failed"
 
 
+class _QUARANTINE_ATR_RETRY_OUTCOMES:
+    ATR_ATTEMPT_PASSED = "dd_quarantine_atr_attempt_passed"
+    ATR_ATTEMPT_FAILED = "dd_quarantine_atr_attempt_failed"
+    ATR_ATTEMPT_SKIPPED = "dd_quarantine_atr_attempt_skipped"
+    ATR_FINAL_PASSED = "dd_quarantine_atr_final_passed"
+    ATR_FINAL_FAILED = "dd_quarantine_atr_final_failed"
+
+
 _FINAL_OUTCOMES: t.Dict[TestStatus, str] = {
     TestStatus.PASS: _ATR_RETRY_OUTCOMES.ATR_FINAL_PASSED,
     TestStatus.FAIL: _ATR_RETRY_OUTCOMES.ATR_FINAL_FAILED,
 }
 
 
+_QUARANTINE_FINAL_OUTCOMES: t.Dict[TestStatus, str] = {
+    TestStatus.PASS: _QUARANTINE_ATR_RETRY_OUTCOMES.ATR_FINAL_PASSED,
+    TestStatus.FAIL: _QUARANTINE_ATR_RETRY_OUTCOMES.ATR_FINAL_FAILED,
+}
+
+
 def atr_handle_retries(
     test_id: InternalTestId,
     item: pytest.Item,
     when: str,
     original_result: pytest_TestReport,
     test_outcome: _TestOutcome,
+    is_quarantined: bool = False,
 ):
+
+    if is_quarantined:
+        outcomes = RetryOutcomes(
+            PASSED=_QUARANTINE_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_PASSED,
+            FAILED=_QUARANTINE_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_FAILED,
+            SKIPPED=_QUARANTINE_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_SKIPPED,
+            XFAIL=_QUARANTINE_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_PASSED,
+            XPASS=_QUARANTINE_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_FAILED,
+        )
+        final_outcomes = _QUARANTINE_FINAL_OUTCOMES
+    else:
+        outcomes = RetryOutcomes(
+            PASSED=_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_PASSED,
+            FAILED=_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_FAILED,
+            SKIPPED=_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_SKIPPED,
+            XFAIL=_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_PASSED,
+            XPASS=_ATR_RETRY_OUTCOMES.ATR_ATTEMPT_FAILED,
+        )
+        final_outcomes = _FINAL_OUTCOMES
+
     # Overwrite the original result to avoid double-counting when displaying totals in final summary
     if when == "call":
         if test_outcome.status == TestStatus.FAIL:
-            original_result.outcome = _ATR_RETRY_OUTCOMES.ATR_ATTEMPT_FAILED
-        return
-    if InternalTest.get_tag(test_id, "_dd.ci.atr_setup_failed"):
-        log.debug("Test item %s failed during setup, will not be retried for Early Flake Detection")
-        return
-    if InternalTest.get_tag(test_id, "_dd.ci.atr_teardown_failed"):
-        # NOTE: tests that passed their call but failed during teardown are not retried
-        log.debug("Test item %s failed during teardown, will not be retried for Early Flake Detection")
```

Comment:
This was a relic from a previous incarnation of the code, these tags are not used anymore; instead this is handled in [_pytest_runtest_makereport](https://github.com/DataDog/dd-trace-py/blob/vitor-de-araujo%2FSDTEST-1255%2Fquarantine/ddtrace/contrib/pytest/_plugin_v2.py#L452) through stashed values.